airflow/executors/kubernetes_executor.py

```diff
@@ -689,9 +689,14 @@ def _change_state(self, key: TaskInstanceKey, state: Optional[str], pod_id: str,
         self.event_buffer[key] = state, None
 
     def try_adopt_task_instances(self, tis: List[TaskInstance]) -> List[TaskInstance]:
-        tis_to_flush = [ti for ti in tis if not ti.queued_by_job_id]
-        scheduler_job_ids = {ti.queued_by_job_id for ti in tis}
-        pod_ids = {ti.key: ti for ti in tis if ti.queued_by_job_id}
+        scheduler_job_ids = {ti.queued_by_job_id for ti in tis if ti.queued_by_job_id}
+
+        # Tasks triggered through API will have no ti.queued_by_job_id
+        # and their pod will have label 'airflow-worker=manual'
+        if any(ti for ti in tis if not ti.queued_by_job_id):
+            scheduler_job_ids.add('manual')
```

Comment:
You are right, not sure how to handle that within Airflow.

Reply:
I think this check allready covers this: https://github.com/apache/airflow/blob/c95081406fa47ee737fa7937be4f13e47b2befca/airflow/executors/kubernetes_executor.py#L717-L734

It would be nice to have a UT for it

Reply:
The `pod_ids` is built from the `tis` and that is constructed with row locks.

Reply:
The `pod_ids` is built from the `tis` and that is constructed with row locks.